gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java

```diff
@@ -378,6 +380,10 @@
               "fs.gs.cooperative.locking.max.concurrent.operations",
               CooperativeLockingOptions.MAX_CONCURRENT_OPERATIONS_DEFAULT);
 
+  /** Configuration key for the headers for HTTP request to GCS. */
+  public static final HadoopConfigurationProperty<Map<String, String>> GCS_HTTP_HEADERS =
+      new HadoopConfigurationProperty<>("fs.gs.http.headers.");
```

Comment:
If this property applies only to GCS requests, better to rename it to `fs.gs.storage.http.headers`